src/rust/ensogl/lib/core/src/display/camera/camera2d.rs

```diff
@@ -165,6 +165,10 @@ struct Camera2dData {
     dirty                  : Dirty,
     zoom_update_registry   : callback::Registry1<f32>,
     screen_update_registry : callback::Registry1<Vector2<f32>>,
+    /// Indicates whether this camera should stay in place.
+    /// Note that this is not enforced by the `Camera` itself, but should be respected by whoever
+    /// modifies the camera.
+    locked                 : bool,
```

Comment:
I don't like this solution. Keeping a flag on camera which is not handled at all by the camera but needs to be handled by something outside, sounds like a bad responsibility design. The camera now keeps `locked` but it is not using this field for anything. This should definitely not be part of a camera implementation.

Reply:
We could use it inside the camera to stop setters from changing the camera when locked, but that seemed overkill. I'm not sure whether there is a better way than just having the status flag.

Reply:
And I wonder if we could apply a workaround here for now and create task for doing it properly?

@wdanilo or have you some another quick and better fix in mind?

Reply:
Discussed during standup: We'll try approach with refactoring Navigator to be a part of Graph Editor view.